PF/Assignment1/Survey.java

```diff
@@ -0,0 +1,127 @@
+//To conduct a survey using open end methodology
+import java.util.Scanner;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class Survey {
+	private static Scanner scan;
+
+	public static void main(String args[]) {
+		try {
+			scan = new Scanner(System.in);
+			BufferedReader br = new BufferedReader(new InputStreamReader(
+					System.in));
+			PrintQuestion object1 = new PrintQuestion();
+			Answer object2 = new Answer();
+			int ch, count = 0, i, result;
+			double totalA = 0.0, totalB = 0.0, totalC = 0.0, totalD = 0.0;
+			char choice;
+			String[] multiChoice = new String[10];
+			String answer = "";
+			String a = ""; // menu drive for user
+			do {
+				System.out.println("Menu");
+				System.out.println("1.Take survey");
+				System.out.println("2.View Statistics");
+				System.out.println("3.View people and answers");
+				System.out.println("Enter your choice");
+				ch = scan.nextInt();
+
+				switch (ch) {
+				case 1:
+					count = count + 1; // To ask survey questions from user
+					System.out.println(object1.singleSelect());
+					result = scan.nextInt();
+					if (result < 0 || result > 5) { // to check for correct
+													// input for single select
+						while (result < 0 || result > 5) {
+							System.out.println("Add correct value(1-4)");
+							result = scan.nextInt();
+							object2.singleAnsSet(result, count);
+						}
+					} else {
+						object2.singleAnsSet(result, count);
+					}
+					System.out.println(object1.multipleSelect()); // to check
+																	// for
+																	// correct
+																	// input of
+																	// multi
+																	// select
+					int j = 0;
+					while ((!answer.equals("#")) && (j < 4)) {
+						answer = br.readLine();
+						if (!answer.equals("1") && !answer.equals("2")
+								&& !answer.equals("3") && !answer.equals("#")) {
+							while (!answer.equals("1") && !answer.equals("2")
+									&& !answer.equals("3")
+									&& !answer.equals("#")) {
+								System.out
+										.println("Add correct value(1-3 or #)");
+								answer = br.readLine();
+								object2.multipleAnsSet(answer, count);
+								j++;
+							}
+						} else {
+
+							a = a + " " + answer;
+							object2.multipleAnsSet(answer, count);
+							j++;
+						}
+					}
+					multiChoice[count] = a;
+					System.out.println(object1.text());
+					answer = br.readLine();
+					;
+					object2.textSet(answer, count);
```

Comment:
ACTION: there is no validation check for text? if user just enter blank space  as text answer?